### PR TITLE
WIP: fix(ecsservice): triggers required when using forceNewDeployment

### PR DESCRIPTION
**Pull request by stooj**
There is a note saying you need to set `forceNewDeployment` when adding `triggers`.

The reverse is also true, so adding a note to say that `triggers` are required when adding `forceNewDeployment`.

### CODE REVIEW

**Comment by stooj** [comment]
Fixes #4162

**Comment by t0yv0** [comment]
Hi @stooj sorry dropped the ball here. @flostadler could you have a look, something's not quite setup in CI to accept docs contributions but if we rerun `make tfgen && make build_sdks` this contrib should be fine I suspect.

**Comment by stooj** [comment]
No worries - I didn't spend the time to set up my local environment for dev contributions (I thought "it's just the docs") but since it's auto-generated I should have checked it locally.

**Comment by t0yv0** [comment]
You're pointing out a real miss we have here, we'd love to remove friction from "just the docs" contributions. I'll talk to the team.

**Comment by flostadler** [comment]
@stooj if you want I can run the re-gen steps. Just lmk

**Comment by stooj** [comment]
@flostadler That'd be great, thanks.

File: provider/cmd/pulumi-resource-aws/schema.json [hunk 1]
```diff
@@ -245414,7 +245414,7 @@
                 },
                 "forceNewDeployment": {
                     "type": "boolean",
-                    "description": "Enable to force a new task deployment of the service. This can be used to update tasks to use a newer Docker image with same image/tag combination (e.g., `myimage:latest`), roll Fargate tasks onto a newer platform version, or immediately deploy `ordered_placement_strategy` and `placement_constraints` updates.\n"
+                    "description": "Enable to force a new task deployment of the service. This can be used to update tasks to use a newer Docker image with same image/tag combination (e.g., `myimage:latest`), roll Fargate tasks onto a newer platform version, or immediately deploy `ordered_placement_strategy` and `placement_constraints` updates.\nWhen using the forceNewDeployment property you also need to configure the Triggers property.\n"
```
Comment (flostadler) [inline provider/cmd/pulumi-resource-aws/schema.json:245417]: Nit: we have two different styles of property naming in the description now. (`placement_constraints` vs `forceNewDeployment`)

**Comment by flostadler** [comment]
@stooj Done!
@t0yv0 @corymhall can you have a look at the changes? Given that I authored parts of the changes I shouldn't review them.

File: provider/doc_edits.go [hunk 2]
```diff
@@ -80,13 +81,25 @@ var fixUpCloudFrontPublicKey = targetedSimpleReplace("cloudfront_public_key.html
...
+	"ecs_service.html.markdown",
+	"* `force_new_deployment` - (Optional) Enable to force a new task deployment of the service. "+
+		"This can be used to update tasks to use a newer Docker image with same image/tag combination "+
+		"(e.g., `myimage:latest`), roll Fargate tasks onto a newer platform version, or immediately deploy "+
+		"`ordered_placement_strategy` and `placement_constraints` updates.",
+	"* `force_new_deployment` - (Optional) Enable to force a new task deployment of the service. "+
+		"This can be used to update tasks to use a newer Docker image with same image/tag combination "+
+		"(e.g., `myimage:latest`), roll Fargate tasks onto a newer platform version, or immediately deploy "+
+		"`ordered_placement_strategy` and `placement_constraints` updates.\n"+
+		"When using the forceNewDeployment property you also need to configure the Triggers property.\n")
```
Comment (t0yv0) [inline provider/doc_edits.go:101]: Is Triggers capitalized intentionally? IN the most used language TypeScript it's probably not capitalized.

File: provider/cmd/pulumi-resource-aws/schema.json (hunk 1 above)
Comment (t0yv0) [reply provider/cmd/pulumi-resource-aws/schema.json:245417]: In my understanding forceNewDeployment is the preferred style when we cannot inflect toward the target language.

**Comment by flostadler** [comment]
Acceptance tests were successful. Bypassing branch protection rules because they're waiting for sentinel

**Pull request merged**